### PR TITLE
TASK: Service provider listing

### PR DESCRIPTION
**Pull request by Sebobo**
Further improvements to layout and functionality

### CODE REVIEW

File: DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Components/MapData.fusion [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            popup = afx`
+                <h4>{serviceProvider.label}</h4>
+                <p @if.set={q(serviceProvider).property('description')}>{q(serviceProvider).property('description')}</p>
+                <address @if.set={q(serviceProvider).property('street')}>
+                    {q(serviceProvider).property('street')}<br/>
+                    {q(serviceProvider).property('zipcode')} {q(serviceProvider).property('city')}
+                </address>
+                <div @if.set={q(serviceProvider).property('website')}>
+                    <i class="fa fa-home"></i>&nbsp;
+                    <a href={q(serviceProvider).property('website')} target="_blank">Website</a>
```
Comment (regniets) [inline DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Components/MapData.fusion]: pls add rel="noopener"

File: DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Components/MapData.fusion [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                </div>
+                <div @if.set={q(serviceProvider).property('email')}>
+                    <i class="fa fa-envelope-square"></i>&nbsp;
+                    <a href={'mailto:' + q(serviceProvider).property('email')}>Email</a>
```
Comment (regniets) [inline DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Components/MapData.fusion]: pls add rel="noopener"

File: DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Documents/ServiceProvider.fusion [hunk 3]
```diff
@@ -1,6 +1,10 @@
 prototype(Neos.NeosIo.ServiceOfferings:Document.ServiceProvider) < prototype(Neos.NeosIo:DefaultPage) {
     body {
         content {
+            keyVisual {
+                keyVisualImage = ${q(documentNode).property('bannerImage')}
```
Comment (regniets) [inline DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Documents/ServiceProvider.fusion]: What happens if no bannerImage is set? Falls back automatically to default?

File: DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Documents/ServiceProvider.fusion [hunk 4]
```diff
@@ -73,85 +85,108 @@ prototype(Neos.NeosIo.ServiceOfferings:Document.ServiceProvider) < prototype(Neo
...
+                                    <h2>Contact</h2>
+
+                                    <ul>
+                                        <li @if.set={props.email}>
+                                            <i class="fa fa-envelope-square"></i>&nbsp;
+                                            <a href={'mailto:' + props.email}>Email</a>
+                                        </li>
+                                        <li @if.set={props.website}>
+                                            <i class="fa fa-home"></i>&nbsp;
+                                            <a href={props.website} target="_blank">Website</a>
```
Comment (regniets) [inline DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Documents/ServiceProvider.fusion]: pls add rel="noopener"

File: DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Documents/ServiceProvider.fusion (hunk 3 above)
Comment (Sebobo) [reply DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Documents/ServiceProvider.fusion]: Yes, just did an override of the property name, everything else is the same.

File: DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Main.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+initMarkerClusters();
+
+const providerListing: HTMLElement = document.getElementById('service-providers-listing');
+
+if (providerListing) {
+    const providerData: Provider[] = JSON.parse(providerListing.dataset.providerData);
+    shuffleArray(providerData);
+
+    console.log(providerData);
```
Comment (regniets) [inline DistributionPackages/Neos.NeosIo.ServiceOfferings/Resources/Private/Fusion/Main.tsx]: For debugging?

**Comment by Sebobo** [comment]
Demo https://vimeo.com/371852265

**Comment by regniets** [comment]
Great work, love it.

**Pull request merged**